### PR TITLE
[Core ML] Update Partitioner Test as Torch.Any Supported

### PR DESCRIPTION
**Pull request by YifanShenSZ**
## Motivation
In https://github.com/pytorch/executorch/pull/4772, the partitioner test case was updated to have graph breaks, because https://github.com/pytorch/pytorch/pull/131863 introduced safe softmax which involves torch.any op that was not supported in Core ML

## Solution
In [coremltools 8.1 release](https://github.com/apple/coremltools/releases/tag/8.1), [torch.any support](https://github.com/apple/coremltools/blob/8.1/coremltools/converters/mil/frontend/torch/ops.py#L1927-L1942) has been included, so
1. We upgrade executorch pinned coremltools to 8.1
2. We will no longer have graph breaks in the partitioner test case

## Other Fixes
https://github.com/pytorch/executorch/issues/5157 and https://github.com/pytorch/executorch/issues/6201 are also resolved in coremltools 8.1

### CODE REVIEW

**Comment by YifanShenSZ** [comment]
@cccclai

File: backends/apple/coreml/scripts/install_requirements.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ rm -rf "$COREML_DIR_PATH/third-party"
 mkdir "$COREML_DIR_PATH/third-party"
 
 echo "${green}ExecuTorch: Cloning coremltools."
-git clone --depth 1 --branch 8.0 "https://github.com/apple/coremltools.git" $COREMLTOOLS_DIR_PATH
+git clone --depth 1 --branch 8.1 "https://github.com/apple/coremltools.git" $COREMLTOOLS_DIR_PATH
```
Comment (cccclai) [inline backends/apple/coreml/scripts/install_requirements.sh:27]: Great!

File: backends/apple/coreml/test/test_coreml_partitioner.py [hunk 2]
```diff
@@ -71,23 +71,15 @@ def test_vit_skip_conv(self):
             )
         )
 
-        conv_block = ["aten.convolution.default", "executorch_call_delegate"]
```
Comment (cccclai) [inline backends/apple/coreml/test/test_coreml_partitioner.py:74]: What does it mean?

**Review by cccclai — APPROVED** [review]
Thanks!

File: backends/apple/coreml/test/test_coreml_partitioner.py (hunk 2 above)
Comment (YifanShenSZ) [reply backends/apple/coreml/test/test_coreml_partitioner.py:74]: In this test case (vit model), vit is fully coreml resident, so we expect the partitioned graph to simply be
```
"aten.convolution.default" -> "executorch_call_delegate" -> "getitem"
```
However, https://github.com/pytorch/pytorch/pull/131863 introduced safe softmax, which involves `torch.any` op that was not supported in coremltools 8.0, so the partitioned graph had 12 blocks separated by `torch.any`

As coremltools 8.1 added `torch.any` support, we get back to the simple no-break graph

**Comment by YifanShenSZ** [comment]
Any legit CI error I should address?

**Comment by cccclai** [comment]
> Any legit CI error I should address?

I think our CI is failing internally but it doesn't show up externally...need to get that fixed than land this diff. cc: @kirklandsign

**Comment by kirklandsign** [comment]
> but it doesn't show up externally

No good workaround for this...

**Comment by YifanShenSZ** [comment]
> @YifanShenSZ this test seems failing
> 
> https://github.com/pytorch/executorch/blob/6ac19cc2d69e34d30668fcd9da9ea5554a6af576/backends/apple/coreml/test/test_coreml_partitioner.py#L59

Passes on my local. What's the error you saw?

**Comment by cccclai** [comment]
Ah acutally I may know why - the internal test probably runs with the old coreml package

**Comment by YifanShenSZ** [comment]
> Ah acutally I may know why - the internal test probably runs with the old coreml package

Hmmm, how come? This PR also updates the pinned coremltools to 8.1?

**Comment by cccclai** [comment]
> > Ah acutally I may know why - the internal test probably runs with the old coreml package
> 
> Hmmm, how come? This PR also updates the pinned coremltools to 8.1, so if CI builds executorch & delegate from source then it should be using the latest?

Internally we pointed to a lower version..@Anthony helped importing the latest coremltools, we should be able to fix it then. So no action on your side, and wait for us :)

**Comment by cccclai** [comment]
hmm it fails to import 
```
    The Pull Request could not be imported cleanly from GitHub. This usually happens for one of two reasons:
    1. The Pull Request is out of date, if you rebase the PR on the latest commit on the GitHub branch and find merge conflicts this is probably the reason
    2. ShipIt is broken for this repo, check bunnylol `oss org/repo`, click on the ShipIt tab, and see if there are any alerts. Fix those and retry the import.
```
Can you rebase?

**Comment by shoumikhin** [comment]
I've already landed this change via the internal coremltools version bump since it failed tests otherwise.
